doc/developer/guide-changes.md

```diff
@@ -0,0 +1,634 @@
+style.
+
+Your release note may sound very similar to your commit message. That's okay!
+
+### PR size limits
+
+As a general rule, the total number of modified lines in a PR (additions plus
+deletions) should not exceed 500 lines. Strive for even smaller PRs when
+possible.
+
+Why are small PRs important? Small PRs:
+
+  * **Are faster and easier to review.** When less code changes, your reviewer
+    will be able to understand it more easily and more quickly.
+
+  * **Get better reviews.** The better your reviewer understands the
+    PR, the better a review you'll get. A reviewer is more able to
+    offer API design suggestions and determine whether the test coverage
+    is complete when the amount of code that is changed is small.
+
+  * **Have fewer merge conflicts.** The more code a PR touches, the
+    more likely it is to rot when other PRs land first.
+
+  * **Are easier to revert.** If a problem is discovered with a PR down the
+    line, it is much easier to revert if it is small and target.
+
+Exceptions:
+
+  * **Generated code.** When generated code is checked into the repository,
+    changes to the generated code can often result in hundreds or thousands
+    of lines of changes. Those changes don't need to count against the limit,
+    provided the reason for the change is clear (e.g., updating the version of
+    the dependency that generates the code).
+
+    Where possible, though, we avoid checking in generated code to the
+    repository
+
+  * **Declarative tests.** Declarative tests, like sqllogictest, testdrive,
+    and datadriven, are in general much easier to review than code. A PR that
+    changes 1000 lines of code is probably acceptable if 750 of the changed
+    lines results are new test cases.
+
+This sentiment from the Google guide is particularly important to keep in mind:
+
+> Keep in mind that although you have been intimately involved with your code
+> from the moment you started to write it, the reviewer often has no context.
+> What seems like an acceptably-sized PR to you might be overwhelming to your
+> reviewer. When in doubt, write PRs that are smaller than you think you need to
+> write. Reviewers rarely complain about getting PRs that are too small.
+
+If you're finding it hard to explain a particular PR in writing, especially for
+large PRs, consider scheduling a brief meeting with your reviewer to walk them
+through the PR. Similarly, as a reviewer, if you have trouble understanding a
+PR, consider scheduling a meeting with your reviewee, rather than continuing to
+communicate by text. The downside is that your discussions won't be recorded for
+posterity, but the increased bandwidth of verbal communication can often make
+the tradeoff worthwhile.
+
+Some PRs, especially large refactors, are nearly impossible to split into
+pieces. If you find yourself in this situation, have a conversation with your
+reviewers ahead of time to get their buy in.
```

Comment:
For larger refactors, I think it makes sense for the engineer to chat with another engineer about their design -- maybe the intended reviewer for the change. This prevents potentially wasted work and an awkward review if the reviewer disagrees with the basic premise of the change.

Reply:
Ack, added a section about early feedback.